components/apimgt/org.wso2.carbon.apimgt.gateway/src/main/java/org/wso2/carbon/apimgt/gateway/handlers/streaming/AsyncAnalyticsDataProvider.java

```diff
@@ -0,0 +1,84 @@
+/*
+ *  Copyright (c) 2021, WSO2 Inc. (http://www.wso2.org) All Rights Reserved.
+ *
+ *  WSO2 Inc. licenses this file to you under the Apache License,
+ *  Version 2.0 (the "License"); you may not use this file except
+ *  in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.wso2.carbon.apimgt.gateway.handlers.streaming;
+
+import org.apache.synapse.MessageContext;
+import org.apache.synapse.core.axis2.Axis2MessageContext;
+import org.apache.synapse.transport.passthru.PassThroughConstants;
+import org.wso2.carbon.apimgt.common.gateway.analytics.exceptions.DataNotFoundException;
+import org.wso2.carbon.apimgt.common.gateway.analytics.publishers.dto.Latencies;
+import org.wso2.carbon.apimgt.common.gateway.analytics.publishers.dto.Operation;
+import org.wso2.carbon.apimgt.common.gateway.analytics.publishers.dto.Target;
+import org.wso2.carbon.apimgt.gateway.handlers.analytics.Constants;
+import org.wso2.carbon.apimgt.gateway.handlers.analytics.SynapseAnalyticsDataProvider;
+import org.wso2.carbon.apimgt.impl.APIConstants;
+
+import static org.wso2.carbon.apimgt.impl.APIConstants.AsyncApi.ASYNC_MESSAGE_TYPE;
+
+public class AsyncAnalyticsDataProvider extends SynapseAnalyticsDataProvider {
+
+    private String apiType;
+    private MessageContext messageContext;
+
+    public AsyncAnalyticsDataProvider(MessageContext messageContext) {
+        super(messageContext);
+        org.apache.axis2.context.MessageContext axisCtx =
+                ((Axis2MessageContext) messageContext).getAxis2MessageContext();
+        apiType = (String) axisCtx.getProperty(PassThroughConstants.SYNAPSE_ARTIFACT_TYPE);
+        this.messageContext = messageContext;
+    }
+
+    @Override
+    public Latencies getLatencies() {
+        return new Latencies();
+    }
+
+    @Override
+    public Operation getOperation() throws DataNotFoundException {
+
+        Operation operation = super.getOperation();
+        String eventName = "";
+        Object eventPrefix = messageContext.getProperty(ASYNC_MESSAGE_TYPE);
```

Comment:
eventPrefix can be empty("")?

Reply:
no, we explicitly set it to a value.

Reply:
if so can we improve it like this?

```
        Operation operation = super.getOperation();
        Object eventPrefix = messageContext.getProperty(ASYNC_MESSAGE_TYPE);
        if (eventPrefix != null) {
            String eventName = eventPrefix.toString();
            operation.setApiResourceTemplate(eventName + operation.getApiResourceTemplate());
        }
        return operation;
```

Reply:
ah yes, already did in https://github.com/wso2/carbon-apimgt/pull/9944/files